.github/actions/cache-node-modules/action.yml

```
@@ -0,0 +1,31 @@
+name: Cache node_modules and dist
+description: Caches the node_modules and optionally dist.
+inputs:
+    shared-dist:
+        description: Whether to cache shared/dist too.
+        default: 'true'
+outputs:
+    cache-hit:
+        description: 'true if node_modules could be restored from cache'
+        value: ${{ steps.cache-node-modules.outputs.cache-hit }}
+
+runs:
+    using: composite
+    steps:
+        - name: Cache node modules
+          uses: actions/cache@v3
+          id: cache-node-modules
+          with:
+              path: |
+                  node_modules
+                  shared/node_modules
+                  frontend/node_modules
+                  backend/node_modules
+              key: node-modules-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('package-lock.json') }}-${{ runner.os }}
```

```suggestion
              key: node-modules-${{ hashFiles('package-lock.json', '**/package-lock.json') }}-${{ runner.os }}
```
https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles